omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java

```diff
@@ -302,7 +303,20 @@ public static void setValue(Obs obs, Object value) throws ParseException, Conver
 			if (obs.getConcept().getDatatype().isCoded()) {
 				// setValueAsString is not implemented for coded obs (in core)
 				Concept valueCoded = (Concept) ConversionUtil.convert(value, Concept.class);
-				obs.setValueCoded(valueCoded);
+				if (valueCoded == null) {
+					//try checking if this this is value drug
+					Drug valueDrug = (Drug) ConversionUtil.convert(value, Drug.class);
+					if (valueDrug != null) {
+						obs.setValueCoded(valueDrug.getConcept());
+						obs.setValueDrug(valueDrug);
+					} else {

```

Comment:
I would instead throw an object not found exception.
The exception message of "value for drug cannot be null" will be confusing for someone who submits a concept uuid of a non existing concept instead of drug but end up getting an error message to do with drugs. Do i make sense?